frontend/src/pages/Polls.js

```diff
@@ -0,0 +1,136 @@
+import Page from "../components/Page";
+import Header from "../components/Header";
+import styled from "styled-components/macro";
+import {useAuth} from "../auth/AuthProvider";
+import {Link, Redirect} from "react-router-dom";
+import Navbar from "../components/Navbar";
+import {useEffect, useState} from "react";
+import {findAll} from "../service/api-service";
+import Loading from "../components/Loading";
+import Error from "../components/Error";
+
+const initialState = [
+    {
+        title: "",
+        givenAnswers: [],
+        possibleAnswers: [],
+        user: {},
+    }
+]
+
+export default function Polls() {
+
+    const {user, token} = useAuth()
+    const [polls, setPolls] = useState(initialState)
+    const [loading, setLoading] = useState(false)
+    const [error, setError] = useState()
+
+    useEffect(() => {
+        if (token) {
+            setError()
+            setLoading(true)
+            findAll(token).then(setPolls).catch(error => {
+                setError(error)
+            })
+            setLoading(false)
+        }
+    }, [token, user])
+
+    if (!user) {
+        return <Redirect to="/"/>
+    }
+
+    return (
+        <Page>
+            <Header title="Polls"/>
+            {loading && <Loading/>}
+            {!loading &&
+            <PollsList>
+                {polls && polls.map(poll => (
```

Comment:
`polls` will always be an array, so you don't have to do the check before the map.